### PR TITLE
Revert and bring back B2 gateway implementation

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Revert and bring back B2 gateway implementation

This PR is simply a revert of 3265112d04c13c2ec49731868824224a0a6af823
just for B2 gateway.

## Motivation and Context
Due to community interest, we are bringing B2 back. 

## Regression
No

## How Has This Been Tested?
Nothing to test, no functionality has changed this is simply a code revert. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |

File: cmd/gateway/b2/gateway-b2.go [hunk 1]
```diff
@@ -0,0 +1,850 @@
+/*
+ * Minio Cloud Storage, (C) 2017, 2018 Minio, Inc.
```
Comment (kannappanr) [inline cmd/gateway/b2/gateway-b2.go:2]: We should add 2019 here.

File: cmd/gateway/b2/gateway-b2_test.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
```
Comment (kannappanr) [inline cmd/gateway/b2/gateway-b2_test.go:2]: We should add 2019 here.

Comment (harshavardhana) [reply cmd/gateway/b2/gateway-b2_test.go:2]: This is a revert not changing anything so will not add 2019

File: cmd/gateway/b2/gateway-b2.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/gateway/b2/gateway-b2.go:2]: This is a revert not changing anything so will not add 2019

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by sinhaashish — APPROVED** [review]
LGTM

**Pull request merged**